### DISCUSSION TITLE
GitHub Action that checks our Actions permissions

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#23016] [Suggest an Idea]
### Tell us more.

It's tricky to be sure if our actions are actually using the least amount of permissions. GitHub released a new beta action that can check our actions. Read the [GitHub Blog, New tool to secure your GitHub Actions](https://github.blog/2023-06-26-new-tool-to-secure-your-github-actions/) to get some background information. 😄 

Here's [the `GitHubSecurityLab/actions-permissions` source]( https://github.com/GitHubSecurityLab/actions-permissions). I recommend you read the:

- root `readme.md`
- `readme.md` in `advisor` directory
- `readme.md` in `monitor` directory

You should also read the code in the repository before trying the action. The action is still in _beta_ so we may want to wait until it's fully cooked.

In any case it sounds like a good thing for us to keep track of. 😉

**Comment by rarkins** [comment]
Sounds like a good idea to try

**Reply by HonkingGoose** [reply]
The action is split into a `advisor` and `monitor` part, the monitor part lists some important limitations. I think the big ones for us are the missing WindowsOS runner support and the missing `discussions` permission support. [^1]

> ## Known limitations
> 
> The Monitor action is not able to detect the usage of the GitHub token in the following cases:
> 
> * WindowsOS runners are not supported (yet, pull-requests are welcome if you know how to redirect all OS outgoing request through a transparent proxy, but not the outgoing proxy traffic itself).
> 
> * GitHub GraphQL API usage is not monitored. It would requires parsing and understanding the GraphQL queries. Pull-requests are welcome.
> 
> * The `discussions` permission type is not supported yet. AFAIK it is GraphQL only.
> 
> * Since the monitor is client based (runs on the Actions runner) it can't detect the usage of the GitHub token if it is passed to a third party web service, which calls the GitHub API back while the Action is running.

[^1]: https://github.com/GitHubSecurityLab/actions-permissions/tree/main/monitor#known-limitations

**Reply by HonkingGoose** [reply]
@zharinov You did a lot of work on our GitHub Action files recently. Do you think the GitHub Action permission checker tool will work for us? Here's the source for the action: [`GitHubSecurityLab/actions-permissions` repository](https://github.com/GitHubSecurityLab/actions-permissions), make sure to read the:

- comment above this one in the thread 😉 
- `README.md` in repository root
- `README.md` in `advisor` directory
- `README.md` in `monitor` directory
